tests/functional/metadata/mongodb/putObject.spec.js

```diff
@@ -0,0 +1,442 @@
+            _replicationConfiguration: null,
+            _lifecycleConfiguration: null,
+            _uid: '',
+            _isNFS: null,
+            ingestion: null,
+        });
+        metadata.createBucket(BUCKET_NAME, bucketMD, logger, err => {
+            if (err) {
+                return done(err);
+            }
+            collection = metadata.client.getCollection(BUCKET_NAME);
+            return done();
+        });
+    });
+
+    afterEach(done => {
+        metadata.deleteBucket(BUCKET_NAME, logger, done);
+    });
+
+    it('Should put a new non versionned object', done => {
+        const bucketName = BUCKET_NAME;
+        const objName = OBJECT_NAME;
+        const objVal = {
+            key: OBJECT_NAME,
+            versionId: 'null',
+            updated: false,
+        };
+        const params = {
+            versioning: null,
+            versionId: null,
+            repairMaster: null,
+        };
+        async.waterfall([
+            next => metadata.putObjectMD(bucketName, objName, objVal, params, logger, err => {
+                if (err) {
+                    return next(err);
+                }
+                return next();
+            }),
+            next => getObject(objName, null, BUCKET_VFORMAT_V0, (err, object) => {
+                assert.deepStrictEqual(err, null);
+                assert.strictEqual(object.key, OBJECT_NAME);
+                return next();
+            }),
+            // When versionning not active only one document is created (master)
+            next => getObjectCount((err, count) => {
+                assert.deepStrictEqual(err, null);
+                assert.strictEqual(count, 1);
+                return next();
+            }),
+        ], done);
+    });
+
+    it('Should update the metadata', done => {
+        const bucketName = BUCKET_NAME;
+        const objName = OBJECT_NAME;
+        const objVal = {
+            key: OBJECT_NAME,
+            versionId: 'null',
+            updated: false,
+        };
+        const params = {
+            versioning: null,
+            versionId: null,
+            repairMaster: null,
+        };
+        async.waterfall([
+            next => metadata.putObjectMD(bucketName, objName, objVal, params, logger, err => {
+                if (err) {
+                    return next(err);
+                }
+                return next();
+            }),
+            next => {
+                objVal.updated = true;
+                metadata.putObjectMD(bucketName, objName, objVal, params, logger, err => {
+                    if (err) {
+                        return next(err);
+                    }
+                    return next();
+                });
+            },
+            // object metadata must be updated
+            next => getObject(objName, null, BUCKET_VFORMAT_V0, (err, object) => {
+                assert.deepStrictEqual(err, null);
+                assert.strictEqual(object.key, OBJECT_NAME);
```

Comment:
we should eiher check against `objName`, or remove that variable and use `OBJECT_NAME` directly everywhere.

```suggestion
                assert.strictEqual(object.key, objName);
```

Reply:
fixed in https://github.com/scality/Arsenal/pull/1729/commits/a6f48b9f342344e6df5ca2e3b7a2d03d8737fbba